_source/_posts/2021-08-06-webforms-blazor-migration.md

```diff
@@ -0,0 +1,839 @@
+    {
+        using (var caloriesContext = new CaloriesContext())
+        {
+            caloriesContext.CalorieDiaries.Remove(new CalorieDiary { Id = id });
+            caloriesContext.SaveChanges();
+        }
+    }
+
+    public List<CalorieDiary> GetDailyList(DateTime date)
+    {
+        using (var caloriesContext = new CaloriesContext())
+        {
+            var list = caloriesContext.CalorieDiaries.Where(diary => diary.AddedAt == date)
+                .Include(d => d.Food).Include(d => d.Exercise)
+                .OrderByDescending(diary => diary.Id).ToList();
+            return list;
+        }
+    }
+}
+
+public class CaloriesContext : DbContext
+{
+    public DbSet<Food> Food { get; set; }
+    public DbSet<Exercise> Exercises { get; set; }
+    public DbSet<CalorieDiary> CalorieDiaries { get; set; }
+
+    protected override void OnConfiguring(DbContextOptionsBuilder options)
+    => options.UseSqlite("Data Source=CaloriesTracker.db");
+
+    protected override void OnModelCreating(ModelBuilder modelBuilder)
+    {
+        //None
+    }
+}
+
+public class CalorieDiary
+{
+    public int Id { get; set; }
+    public DateTime AddedAt { get; set; }
+
+    public int? FoodId { get; set; }
+    public Food Food { get; set; }
+
+    public int? ExerciseId { get; set; }
+    public Exercise Exercise { get; set; }
+}
+
+public class Food
+{
+    public int Id { get; set; }
+    public string Name { get; set; }
+    public int Calories { get; set; }
+}
+
+public class Exercise
+{
+    public int Id { get; set; }
+    public string Name { get; set; }
+    public int Calories { get; set; }
+}
+```
+
+Now, let's migrate the project to Blazor. First, I will create a new Blazor project.
+
+## Create a Blazor Project
+
+Fire up Visual Studio, create a new project by clicking **File>New Project**, select **Blazor App**, and click **Next**. Name the app *CaloriesTracker.Blazor* and click *Create*.
+
+{% img blog/webforms-blazor-migration/image2.png alt:"" width:"800" %}{: .center-image }
+
+In the next window, select **Blazor Server App**. Now, click **Create**, and Visual Studio will create a new server-side Blazor project.
+
+{% img blog/webforms-blazor-migration/image3.png alt:"" width:"800" %}{: .center-image }
+
+Alternatively, you can create the project by running the following command in a command shell:
+
+```sh
+dotnet new blazorserver -o CaloriesTracker.Blazor
+```
+
+Blazor Server hosts Razor components on the server in an ASP.NET Core app and handles UI updates over a SignalR connection. The Blazor runtime sends events from browser to server and applies UI updates to the components that the server sends back. This strategy is similar to the approach used by ASP.NET AJAX `UpdatePanel` control that handles partial page updates in response to trigger events. The main difference is that Blazor Server maintains an active connection to the server, and all the UI state stays on the server instead of `ViewState`.
```

Comment:
```suggestion
Blazor Server hosts Razor components on the server in an ASP.NET Core app and handles UI updates over a SignalR connection. The Blazor runtime sends events from browser to server and applies UI updates to the components that the server sends back. This strategy is similar to the approach used by the ASP.NET AJAX `UpdatePanel` control that handles partial page updates in response to trigger events. The main difference is that Blazor Server maintains an active connection to the server, and all the UI state stays on the server instead of `ViewState`.
```
missing article for "control"